store/query.go

```diff
@@ -289,19 +290,26 @@ func (r *Result) ToMap() (data map[string]interface{}) {
 	data = make(map[string]interface{})
 	data["id"] = r.Id
 	data["kind"] = r.Kind
-	var ts int64
+	var ts_latest int64
+	var ts_oldest int64
+	ts_oldest = time.Now().UnixNano()
 	for pred, versions := range r.Columns {
 		// During conversion to JSON, to keep things simple,
 		// we're dropping older versions of predicates, and
 		// source and ts information across all the predicates,
 		// keeping only the latest one.
 		data[pred] = versions.Latest().Value
-		data["source"] = versions.Latest().Source // Loss of information.
-		if versions.Latest().NanoTs > ts {
-			ts = versions.Latest().NanoTs
+		if versions.Latest().NanoTs > ts_latest {
+			ts_latest = versions.Latest().NanoTs
+			data["source"] = versions.Latest().Source // Loss of information.
+		}
+		if versions.Oldest().NanoTs < ts_oldest {
+			ts_oldest = versions.Oldest().NanoTs
+			data["creator"] = versions.Oldest().Source
 		}
 	}
-	data["ts_millis"] = int(ts / 1000000) // Loss of information. Picking up latest mod time.
+	data["creation_ts"] = int(ts_oldest / 1000000)
+	data["ts_millis"] = int(ts_latest / 1000000) // Loss of information. Picking up latest mod time.
```

Comment:
Replace the "_ts" with "_ms", to represent milliseconds.
data["modification_ms"]
data["creation_ms"]